source/server/instance_impl.cc

```diff
@@ -11,8 +11,8 @@ void InstanceImpl::maybeCreateHeapShrinker() {
       std::make_unique<Memory::HeapShrinker>(dispatcher(), overloadManager(), *stats().rootScope());
 }
 
-std::unique_ptr<OverloadManager> InstanceImpl::createOverloadManager() {
-  return std::make_unique<OverloadManagerImpl>(
+absl::StatusOr<std::unique_ptr<OverloadManager>> InstanceImpl::createOverloadManager() {
```

Comment:
nit: maybe we should add a alias for `std::unique_ptr<OverloadManager>` as done for other unique ptrs. Will reduce the nesting quite a bit.

Reply:
I'm going to leave as is as I'm on a team of engineers who strongly dislike the FooPtr paragim =P